src/main/java/org/jenkinsci/plugins/codesonar/conditions/WarningCountIncreaseSpecifiedScoreAndHigherCondition.java

```diff
@@ -67,48 +64,29 @@ public void setWarrantedResult(String warrantedResult) {
     }
 
     @Override
-    public Result validate(CodeSonarBuildActionDTO current, CodeSonarBuildActionDTO previous, Launcher launcher, TaskListener listener, CodeSonarLogger csLogger) {
+    public Result validate(CodeSonarHubAnalysisDataLoader current, CodeSonarHubAnalysisDataLoader previous, Launcher launcher, TaskListener listener, CodeSonarLogger csLogger) throws CodeSonarPluginException {
         if (current == null) {
             registerResult(csLogger, CURRENT_BUILD_DATA_NOT_AVAILABLE);
             return Result.SUCCESS;
         }
         
-        Analysis analysis = current.getAnalysisActiveWarnings();
-        
-        // Going to produce build failure in the case of missing necessary information
-        if(analysis == null) {
-            LOGGER.log(Level.SEVERE, "\"analysisActiveWarnings\" data not found in persisted build.");
-            registerResult(csLogger, CURRENT_BUILD_DATA_NOT_AVAILABLE);
-            return Result.FAILURE;
-        }
-
-        int totalNumberOfWarnings = analysis.getWarnings().size();
-
-        int severeWarnings = 0;
-        List<Warning> warnings = analysis.getWarnings();
-        for (Warning warning : warnings) {
-            if (warning.getScore() > rankOfWarnings) {
-                severeWarnings++;
-            }
+        if (previous == null) {
+            registerResult(csLogger, PREVIOUS_BUILD_DATA_NOT_AVAILABLE);
+            return Result.SUCCESS;
         }
         
-        float calculatedWarningPercentage;
-        //If there are no warnings, redefine percentage of new warnings
-        if(totalNumberOfWarnings == 0) {
-            calculatedWarningPercentage = severeWarnings > 0 ? 100f : 0f;
-            LOGGER.log(Level.INFO, "no warnings found, forcing severe warning percentage to {0,number,0.00}%", calculatedWarningPercentage);
-        } else {
-            calculatedWarningPercentage = ((float) severeWarnings / totalNumberOfWarnings) * 100;
-            LOGGER.log(Level.INFO, "severe warnings percentage = {0,number,0.00}%", calculatedWarningPercentage);
-        }
-
+        long warningsAboveThresholdForCurrent = current.getNumberOfWarningsWithScoreAboveThreshold(rankOfWarnings);
+        long warningsAboveThresholdForPrevious = previous.getNumberOfWarningsWithScoreAboveThreshold(rankOfWarnings);
+        
         float thresholdPercentage = Float.parseFloat(warningPercentage);
+        float calculatedWarningPercentage = ((float) warningsAboveThresholdForCurrent / warningsAboveThresholdForPrevious) * 100;;
```

Comment:
What if `warningsAboveThresholdForPrevious` is zero?  Is "NaN" or "Inf" an acceptable response here?

Probably we should pretend that the previous count was 1, not zero.  This way if, say, 1 new warning above threshold were added and there were none previously, the result would be 100%.  Maybe that is not technically accurate, but I think it better fits the intention of this condition.

Reply:
Also, this doesn't calculate an "increase", it calculates a simple ratio.  Probably the formula should be `100*(c-p)/p` when p is not zero.

See issue #17, which includes an excerpt from the documentation.

Reply:
I updated the code to try and fix the computation and the zero denominator case.  I think the behavior more closely matches then intention and the documentation now.